# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 270961083337
# Full Hash: 270961083337349dc71f3a2359b7cdc4b1c83a0f
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r db39bf2058e1 -r 270961083337 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Oct 20 10:24:40 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Oct 20 10:24:47 2020 +0000
@@ -4419,12 +4419,11 @@
   explicit WasmABIArgIter(const VecT& types)
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
-
 }  // namespace jit
 
 namespace wasm {
-TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp);
-TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp);
+const TlsData* ExtractCalleeTlsFromFrameWithTls(const Frame* fp);
+const TlsData* ExtractCallerTlsFromFrameWithTls(const Frame* fp);
 }  // namespace wasm
 
 }  // namespace js